src/annotator/config/test/settings-test.js

```diff
@@ -267,69 +267,97 @@ describe('annotator.config.settingsFrom', function() {
   });
 
   describe('#hostPageSetting', function() {
-    context('when the client is from a browser extension', function() {
-      beforeEach('configure a browser extension client', function() {
-        fakeIsBrowserExtension.returns(true);
-      });
-
-      it('always returns null', function() {
-        // These settings in the host page should be ignored.
-        fakeConfigFuncSettingsFrom.returns({foo: 'bar'});
-        fakeSharedSettings.jsonConfigsFrom.returns({foo: 'bar'});
-
-        assert.isNull(settingsFrom(fakeWindow()).hostPageSetting('foo'));
-      });
-    });
-
-    context('when the client is embedded in a web page', function() {
-      beforeEach('configure an embedded client', function() {
-        fakeIsBrowserExtension.returns(false);
-      });
-
-      it('returns setting values from window.hypothesisConfig()', function() {
-        fakeConfigFuncSettingsFrom.returns({foo: 'bar'});
-
-        assert.equal(settingsFrom(fakeWindow()).hostPageSetting('foo'), 'bar');
-      });
-
-      it('returns setting values from js-hypothesis-config scripts', function() {
-        fakeSharedSettings.jsonConfigsFrom.returns({foo: 'bar'});
-
-        assert.equal(settingsFrom(fakeWindow()).hostPageSetting('foo'), 'bar');
-      });
-
-      specify('hypothesisConfig() overrides js-hypothesis-config', function() {
-        fakeConfigFuncSettingsFrom.returns({
-          foo: 'fooFromHypothesisConfig',
-        });
-        fakeSharedSettings.jsonConfigsFrom.returns({
-          foo: 'fooFromJsHypothesisConfigScript',
-        });
-
-        assert.equal(
-          settingsFrom(fakeWindow()).hostPageSetting('foo'),
-          'fooFromHypothesisConfig'
-        );
-      });
-
-      [
-        null,
-        undefined,
-      ].forEach(function(returnValue) {
-        specify('even a ' + returnValue + ' from hypothesisConfig() overrides js-hypothesis-configs', function() {
-          fakeConfigFuncSettingsFrom.returns({foo: returnValue});
-          fakeSharedSettings.jsonConfigsFrom.returns({foo: 'bar'});
-
-          assert.equal(settingsFrom(fakeWindow()).hostPageSetting('foo'), returnValue);
+    [
+      {
+        when: 'the client is embedded in a web page',
+        specify: 'it returns setting values from window.hypothesisConfig()',
+        isBrowserExtension: false,
```

Comment:
These same `when` and `isBrowserExtension` values here are repeated several times, but I think I'm okay with that duplication for simplicity's sake